Source/CocoaMQTT.swift

```diff
@@ -490,18 +502,45 @@ extension CocoaMQTT: GCDAsyncSocketDelegate {
     }
 
     public func socketDidDisconnect(_ sock: GCDAsyncSocket, withError err: Error?) {
+        // Clean up
         socket.delegate = nil
         connState = .disconnected
         delegate?.mqttDidDisconnect(self, withError: err)
         didDisconnect(self, err)
-
-        autoReconnTimer = nil
+        
         if disconnectExpectedly {
             connState = .initial
-        } else if autoReconnect && autoReconnectTimeInterval > 0 {
+        } else if autoReconnect {
+            
+            // Generate exponential time point array
+            var exponential = autoReconnectTimeInterval
+            while(true) {
```

Comment:
Hi @HJianBo, the disconnect gets called on every unsuccessful retry and the exponential array is flooded with multiple repetitions of same set of values `[1 to maxAutoReconnectTimeInterval]`.

Reply:
Yep, it's a bug. The exponential array should be generated once time only.  I'll fixed it later